src/Logger-MCC_DAQ/main.py

```diff
@@ -0,0 +1,196 @@
+# Required Third Party Libraries, which are included in the Device Class
+# package for convenience purposes, may have a different license. You can
+# find those in the corresponding folders or contact the maintainer.
+#
+# MIT License
+#
+# Copyright (c) 2023 SweepMe! GmbH (sweep-me.net)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# Contribution: We like to thank TU Dresden/Shayan Miri for providing the initial version of this driver.
+
+# SweepMe! device class
+# Type: Logger
+# Device: Measurement Computing Corporation DAQ devices
+
+from pysweepme import addFolderToPATH
+from pysweepme.ErrorMessage import error, debug
+from pysweepme.EmptyDeviceClass import EmptyDevice
+
+addFolderToPATH()
+
+# this driver needs libraries installed by the manufacturer software InstaCal
+try:
+    from mcculw import ul
+    from mcculw.ul import ULError
+    from mcculw.device_info import DaqDeviceInfo
+    from mcculw.enums import ScanOptions, FunctionType, Status, AnalogInputMode, InterfaceType, ULRange
+except FileNotFoundError as e:
+    msg = "MCC DAQ Software missing. Install InstaCal and Universal Library (UL)."
+    raise ImportError(msg) from e
+
+
+class Device(EmptyDevice):
+    description = """
+                  MCC High-Speed Multifunction DAQ
+                  
+                  To use this driver, installation of MCC DAQ Software, including Universal Library™ is needed. 
+                  Please download it here: https://www.mccdaq.com/Software-Downloads
+                  
+                  If your device supports additional AI ranges, they can be added by extending the available_ai_ranges 
+                  dictionary.
+                  """
+
+    def __init__(self):
+
+        EmptyDevice.__init__(self)
+
+        self.shortname = "MCC-DAQ"  # short name will be shown in the sequencer
+        self.variables = []
+        self.units = []
+
+        self.port_manager = False
```

Comment:
Because the port manager is switched off by default, this line can be removed. I think it is a leftover from the driver that was contributed.

https://github.com/SweepMe/pysweepme/blob/ada2b1ecfb8bad095862eb78046ab9178ea30bf1/src/pysweepme/EmptyDeviceClass.py#L65